### PR TITLE
Add preliminary support for `exe` packaging type

### LINKED ISSUE #907: `packaging` maven coordinate doesn't seem to be honored
The documentation mentions that:
https://github.com/bazelbuild/rules_jvm_external/blob/643f9c9c8ec7bb5271a3523c07e7d19e93dab238/README.md?plain=1#L421-L422

But, when passing something like `com.google.protobuf:protoc:exe:linux-x86_64:3.21.12`, the `exe` packaging seems to be ignored:
```
Error in fail: Error while fetching artifact with coursier: Error fetching artifacts:
https://repo1.maven.org/maven2/com/google/protobuf/protoc/3.21.12/protoc-3.21.12-linux-x86_64.jar: not found: https://repo1.maven.org/maven2/com/google/protobuf/protoc/3.21.12/protoc-3.21.12-linux-x86_64.jar
```
(instead of `https://repo1.maven.org/maven2/com/google/protobuf/protoc/3.21.12/protoc-3.21.12-linux-x86_64.exe`)

Did I miss something?

### PR DESCRIPTION
**Pull request by rdesgroppes**
Resolves #907.

This is to allow fetching prebuilt binaries (such as [protoc](https://repo1.maven.org/maven2/com/google/protobuf/protoc/)) in the same way as other artifacts served by maven repositories.

Background: particularly useful for defining (lang) proto toolchains, without having to compile nor manually maintain repository mappings (checksums, etc.) for all supported platforms.

A few related questions/contributions:
- #132
- https://github.com/bazelbuild/rules_proto/issues/35
- https://github.com/bazelbuild/rules_proto/pull/36
- https://github.com/bazelbuild/rules_proto/pull/83
- https://github.com/bazelbuild/rules_proto/pull/107
- https://github.com/bazelbuild/rules_proto/pull/113
- https://github.com/bazelbuild/rules_proto/pull/157
- https://github.com/rules-proto-grpc/rules_proto_grpc/issues/88
- https://github.com/stackb/rules_proto#workspace-boilerplate (`@build_stack_rules_proto//toolchain:prebuilt`)

### CODE REVIEW

**Comment by rdesgroppes** [comment]
Hi @cheister, may I kindly request a review?

**Comment by rdesgroppes** [comment]
Or @shs96c, maybe?

File: WORKSPACE [hunk 1]
```diff
@@ -253,6 +253,10 @@ maven_install(
             group = "com.google.apis",
             version = "v1-rev235-1.25.0",
         ),
+        # https://github.com/bazelbuild/rules_jvm_external/issues/907
+        # Any two platforms to ensure that it doesn't work _only_ under the host operating system
```
Comment (shs96c) [inline WORKSPACE:257]: Excellent. Thank you

**Review by shs96c — APPROVED** [review]
The CI tests are passing, and this makes sense. Thank you for taking the time to put this PR together, and your patience waiting for a review!

File: WORKSPACE (hunk 1 above)
Comment (rdesgroppes) [reply WORKSPACE:257]: I do thank _you_, @shs96c, knowing how time-consuming code reviews can be...

**Pull request merged**